GUI/Types/Renderer/Shaders/post_processing.frag

```diff
@@ -0,0 +1,125 @@
+#version 460
+
+#include "common/utils.glsl"
+
+//#define F_OUTPUT_LINEAR 0
+#define F_COLOR_CORRECTION_LUT 0
+#define F_HAS_BLUE_NOISE 0
+#define F_IS_POST_HLA 0
+//#define F_BLOOM 0
+//#define F_OLD_TONEMAPPING_CURVE 0 // HLA, linear only. Used in linear viewmodes to disable the tonemapping curve
+
+// Bloom might eventually be supported
+//uniform sampler2D g_tBloom; // t13
+//uniform vec4 g_vBloomUvScaleClamp;
+//uniform vec3 g_vNormalizedBloomStrengths;
+//uniform vec3 g_vUnNormalizedBloomStrengths;
+
+
+
+
+layout(location=0) uniform sampler2DMS g_tColorBuffer;
+
+vec4 SampleColorBuffer(vec2 coords)
+{
+    vec4 singleSampleColor = texelFetch(g_tColorBuffer, ivec2(coords.xy), int(gl_SampleID));
+    singleSampleColor.rgb = singleSampleColor.rgb / (max3(singleSampleColor.rgb) + 1.0);
+    return singleSampleColor;// / 2.8;
+/*
+// This pretty much implements msaa_resolve_cs as a pixel shader, and without the 4x downsample
+    vec4 msaaResolveColor = vec4(0.0);
+    float INV_SAMPLES = 1.0 / F_MSAA_SAMPLES;
+
+    for (uint i = uint(F_MSAA_SAMPLES); i < F_MSAA_SAMPLES; i++)
+    {
+        vec4 singleSampleColor = texelFetch(g_tColorBuffer, ivec2(coords.xy), int(i));
+        singleSampleColor = clamp(singleSampleColor, vec4(0.0), vec4(65504.0));
+
+        msaaResolveColor += singleSampleColor;
+    }
+    return msaaResolveColor;
+*/
+}
+
+
+
+uniform float g_flToneMapScalarLinear;
+uniform float g_flExposureBiasScaleFactor;
+uniform float g_flShoulderStrength;
+uniform float g_flLinearStrength;
+uniform float g_flLinearAngle;
+uniform float g_flToeStrength;
+uniform float g_flToeNum;
+uniform float g_flToeDenom;
+uniform float g_flWhitePointScale;
+
+vec3 TonemapColor(vec3 vColor)
+{
+    vColor *= (g_flToneMapScalarLinear * g_flExposureBiasScaleFactor) * 2.8;
+
+    // Uncharted tonemapper
+    vec3 tonemapNumerator = vColor * (g_flShoulderStrength * vColor + g_flLinearStrength * g_flLinearAngle) + g_flToeNum * g_flToeStrength;
+    vec3 tonemapDenominator = vColor * (g_flShoulderStrength * vColor + g_flLinearStrength) + g_flToeDenom * g_flToeStrength;
+
+    vec3 vTonemappedColor = tonemapNumerator / tonemapDenominator - g_flToeNum / g_flToeDenom;
+
+    // Divide by tonemapped white point (pre-calculated on CPU)
+    vTonemappedColor *= g_flWhitePointScale; // This is actually 1/TonemapColor(WhitePoint)
+
+    // Finally, convert from Linear to Gamma space
+    vec3 vTonemappedColorSRGB = SrgbLinearToGamma(vTonemappedColor);
+
+    return vTonemappedColorSRGB;
+}
+
+
+
+#if F_COLOR_CORRECTION_LUT
+
+layout(location=2) uniform sampler3D g_tColorCorrection;
+uniform float g_flColorCorrectionDefaultWeight;
+
+vec3 ApplyColorCorrection(vec3 vColor)
+{
+    vec3 ColorCorrectedColor = texture(g_tColorCorrection, saturate(vColor) * 0.9688 + 0.0156).rgb;
```

Comment:
```suggestion
    uniform int g_nColorCorrectionSize = 32;
    const float x = (g_nColorCorrectionSize-1/g_nColorCorrectionSize);
    const float y = (1/(g_nColorCorrectionSize*2);
    vec3 ColorCorrectedColor = texture(g_tColorCorrection, saturate(vColor) * x +y).rgb;
```

Reply:
Good point- funnily enough this is based on Valve's shader code, so I guess they just assumed it'll always be 32

Reply:
That might have been optimized by the compiler.

Reply:
Yeah weird that its hardcoded, but we can do better why not.